app/src/main/java/com/blogspot/e_kanivets/moneytracker/activity/ReportActivity.kt

```diff
@@ -0,0 +1,97 @@
+package com.blogspot.e_kanivets.moneytracker.activity
+
+import android.view.View
+import android.widget.AdapterView
+import android.widget.ArrayAdapter
+import com.blogspot.e_kanivets.moneytracker.R
+import com.blogspot.e_kanivets.moneytracker.activity.base.BaseBackActivity
+import com.blogspot.e_kanivets.moneytracker.adapter.ExpandableListReportAdapter
+import com.blogspot.e_kanivets.moneytracker.controller.CurrencyController
+import com.blogspot.e_kanivets.moneytracker.controller.data.ExchangeRateController
+import com.blogspot.e_kanivets.moneytracker.controller.data.RecordController
+import com.blogspot.e_kanivets.moneytracker.entity.Period
+import com.blogspot.e_kanivets.moneytracker.entity.data.Record
+import com.blogspot.e_kanivets.moneytracker.report.ReportMaker
+import com.blogspot.e_kanivets.moneytracker.report.record.RecordReportConverter
+import com.blogspot.e_kanivets.moneytracker.ui.presenter.ShortSummaryPresenter
+import kotlinx.android.synthetic.main.activity_report.*
+import javax.inject.Inject
+
+class ReportActivity : BaseBackActivity() {
+
+    @Inject
+    lateinit var recordController: RecordController
+    @Inject
+    lateinit var rateController: ExchangeRateController
+    @Inject
+    lateinit var currencyController: CurrencyController
+
+    private lateinit var recordList: List<Record>
+    private var period: Period? = null
+
+    private lateinit var shortSummaryPresenter: ShortSummaryPresenter
+
+    override fun getContentViewId(): Int {
```

Comment:
Can be simplified to `override fun getContentViewId() = R.layout.activity_report`.